### PR TITLE
UCP/RNDV: Fix "rndv through am" thresholds definition

### PR DESCRIPTION
**Pull request by ivankochin**
## What
Allows to define RNDV/AM scheme via RNDV_SCHEME control and applies this new option to protocols. 

## Why ?
Now RNDV/AM protos can be chosen even if RNDV_SCHEME is defined to different value.

### CODE REVIEW

**Comment by yosefe** [comment]
Seems this PR reveals existing bug - "memtrack.c:329  UCX  WARN  allocated zero-size block 0x615a9d0 for dt_iov_memh"

File: src/ucp/core/ucp_context.c [hunk 1]
```diff
@@ -223,6 +223,7 @@ static ucs_config_field_t ucp_context_config_table[] = {
    "Communication scheme in RNDV protocol.\n"
    " get_zcopy - use get_zcopy scheme in RNDV protocol.\n"
    " put_zcopy - use put_zcopy scheme in RNDV protocol.\n"
+   " am        - use am scheme in RNDV protocol.\n"
```
Comment (tvegas1) [inline src/ucp/core/ucp_context.c:226]: Maybe remove that line as #9154 is already taking of this?

Comment (ivankochin) [reply src/ucp/core/ucp_context.c]: Done, tnx!

**Review by yosefe — APPROVED** [review]

File: src/ucp/dt/datatype_iter.c [hunk 2]
```diff
@@ -115,7 +119,7 @@ void ucp_datatype_iter_iov_mem_dereg(ucp_datatype_iter_t *dt_iter)
     ucp_mem_h *memh = dt_iter->type.iov.memh;
     size_t iov_index, length;
 
-    if ((memh == NULL) || ucp_memh_is_user_memh(*memh)) {
+    if ((memh == NULL) || (*memh == NULL) || ucp_memh_is_user_memh(*memh)) {
```
Comment (brminich) [inline src/ucp/dt/datatype_iter.c:122]: pls check when we can have `memh != NULL` but `*memh ==NULL`

File: src/ucp/dt/datatype_iter.c [hunk 3]
```diff
@@ -78,7 +82,7 @@ ucs_status_t ucp_datatype_iter_iov_mem_reg(ucp_context_h context,
     ucs_status_t status;
     size_t iov_index;
 
-    if (md_map == 0) {
+    if (md_map == 0 || iov_count == 0) {
```
Comment (brminich) [inline src/ucp/dt/datatype_iter.c:85]:
```suggestion
    if ((md_map == 0) || (iov_count == 0)) {
```

File: src/ucp/dt/datatype_iter.c (hunk 2 above)
Comment (ivankochin) [reply src/ucp/dt/datatype_iter.c:122]: With the new `iov_count` checks it seems like there is no such case. Thanks!

```suggestion
    if ((memh == NULL) || ucp_memh_is_user_memh(*memh)) {
```

**Review by brminich — APPROVED** [review]

**Review by yosefe — APPROVED** [review]

**Comment by ivankochin** [comment]
Squashed, @brminich @yosefe pls re-approve

**Review by brminich — APPROVED** [review]

**Pull request merged**